src/types/string.rs

```diff
@@ -212,32 +212,48 @@ impl PyString {
     /// Attempts to create a Python string from a Python [bytes-like object].
     ///
     /// [bytes-like object]: (https://docs.python.org/3/glossary.html#term-bytes-like-object).
+    pub fn from_encoded_object<'py>(
+        src: &Bound<'py, PyAny>,
+        encoding: &CStr,
+        errors: &CStr,
```

Comment:
Should these be `Option`s to ["have the interface use the default values"](https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_FromEncodedObject) or do we want to explicitly not provide that option?

Reply:
Ah good spot. I'll have a think about this, I'm not sure which I prefer 🤔

Reply:
I decided I liked this idea and added the `Option<>` wrappers 👍